userspace/libsinsp/sinsp.cpp

```diff
@@ -1392,14 +1392,21 @@ uint64_t sinsp::get_num_events()
 
 sinsp_threadinfo* sinsp::find_thread_test(int64_t tid, bool lookup_only)
 {
-	return find_thread(tid, lookup_only);
+	// TODO: we pay the refcount manipulation price here
+	return &*find_thread(tid, lookup_only);
 }
 
 sinsp_threadinfo* sinsp::get_thread(int64_t tid, bool query_os_if_not_found, bool lookup_only)
 {
-	sinsp_threadinfo* sinsp_proc = find_thread(tid, lookup_only);
+	// TODO: we pay the refcount manipulation price here
+	return &*get_thread_ref(tid, query_os_if_not_found, lookup_only);
```

Comment:
This function is unsafe and should returned a shared_ptr instead. There's no guaranteed that the raw ptr we return is valid after the function return.